### ISSUE TITLE
[Due for payment 2025-03-05] [$250] Expense - "No further action required" message, doesn´t disappear after cancelling payment.

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#54518]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.78-1
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/5367571&group_by=cases:section_id&group_order=asc&group_id=296775
**Email or phone of affected tester (no customers):** ibellicotest+115@gmail.com
**Issue reported by:** Applause Internal Team
**Device used:** Motorola MotoG60 - Android 12 - Chrome / Windows 10 - Chrome
**App Component:** Search

## Action Performed:
1. Open the staging.new.expensify.com website.
2. Open a workspace chat and submit an expense to it.
3. Go to "Reports" section.
4. Open the expense.
5. Pay the expense with "Pay Elsewhere" option.
6. Verify that a "No further action required" message is displayed on top of expense report.
7. Tap on report header.
8. Tap on "Cancel Payment"
9. Return to expense report

## Expected Result:
After the payment is cancelled, we should go back a step and show the same message that appeared before clicking "pay elsewhere"

## Actual Result:
"No further action requiered" remains visible on expense report after the expense payment is cancelled.

## Workaround:
Unknown

## Platforms:
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/9d3e4c2b-f692-4d31-a5c2-1135fc69a8a2

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021874212219304994702</li>
        <li>Upwork Job ID: 1874212219304994702</li>
        <li>Last Price Increase: 2025-01-07</li>
<li>Automatic offers: </li>
<ul>
<li>daledah | Contributor | 105611947</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @akinwale</details>

**Comment by Shahidullah-Muffakir** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
"No further action required" message, doesn´t disappear after cancelling payment
### What is the root cause of that problem?
We did not consider this case here:
https://github.com/Expensify/App/blob/dc784de259688ac6891c808502347fb231aeca94/src/libs/NextStepUtils.ts#L83
### What changes do you think we should make in order to solve the problem?

We can add one more check if the `isCancelledIOU` is true, return:
 ```
if (!ReportUtils.isExpenseReport(report) || report?.isCancelledIOU) {
        return null;
    }
```
### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?

**Comment by akinwale** [comment]
@Shahidullah-Muffakir's [proposal](https://github.com/Expensify/App/issues/54518#issuecomment-2561060424) looks good here.

🎀👀🎀 C+ reviewed.

**Comment by stitesExpensify** [comment]
@daledah is that what used to happen? As in do we have code that is no longer being hit? I am asking because the issue description specifically says it should just not be shown

**Comment by Shahidullah-Muffakir** [comment]
I agree with @daledah. The expected behavior in the original issue isn’t fully explained. When a user cancels the payment, the report should go back to how it was before the "Pay Elsewhere" option was clicked,  the same as if the payment step never happened.

Also, I’ve tested this issue on both prod and staging, and I wasn’t able to reproduce it.

**Comment by stitesExpensify** [comment]
Okay thank you very much for your help.

@akinwale Can you please choose a new proposal given the new solution so that we can move forward?

**Comment by anmurali** [comment]
Waiting for @akinwale to pick a new proposal

**Comment by akinwale** [comment]
In this case, we can move forward with @daledah's [proposal](https://github.com/Expensify/App/issues/54518#issuecomment-2561591869) instead.

cc @stitesExpensify

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by akinwale** [comment]
@daledah Any updates on the PR? Thanks.

**Comment by akinwale** [comment]
@daledah That looks fine. You can proceed.

**Comment by akinwale** [comment]
@daledah Yes, this is correct.

**Comment by stitesExpensify** [comment]
That looks correct to me

**Comment by anmurali** [comment]
@daledah are you good to move forward?

**Comment by anmurali** [comment]
@daledah do you have an update on this PR?

**Comment by akinwale** [comment]
# BugZero Checklist:

- [x] **[@akinwale]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [x] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [x] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [x] 3d. QA
  - [ ] 3z. Other:

</details>

- [x] **[@akinwale]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment:

- [x] **[@akinwale]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion:

- [x] **[@akinwale]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal

### Test:

1. Launch Expensify.
2. Open a workspace chat report and submit an expense.
3. Navigate to the expense report as the workspace owner / approver.
4. Pay the expense with the Pay Elsewhere option.
5. Tap on the report header.
6. Select Cancel Payment and then confirm.
7. Verify that the message displayed in the header is the same as the message that was previously displayed before paying the expense.

Do we agree 👍 or 👎?

**Comment by anmurali** [comment]
- [ ] @akinwale $250 approved NewDot Manual Requests
- [X] @daledah $250 approved [automatic offer](https://www.upwork.com/nx/wm/offer/105611947) (Contributor)

**Comment by JmillsExpensify** [comment]
$250 approved for @akinwale
